src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -885,5 +887,42 @@ private static string GetEnvironmentVariablesBlock(IDictionary<string, string> s
         }
 
         private static string GetErrorMessage(int error) => Interop.Kernel32.GetMessage(error);
+
+        /// <devdoc>
+        ///    <para>Gets the friendly name of the process.</para>
+        /// </devdoc>
+        public string ProcessName
+        {
+            get
+            {
+                EnsureState(State.HaveNonExitedId);
+
+                if (_processName == null)
+                {
+                    // If we already have the name via a populated ProcessInfo
+                    // then use that one.
+                    if (_processInfo?.ProcessName != null)
+                    {
+                        _processName = _processInfo!.ProcessName;
+                    }
+                    else
+                    {
+
+                        // If we don't have a populated ProcessInfo, then create and
+                        // cache the process name.
+                        EnsureState(State.HaveId);
+
+                        _processName = ProcessManager.GetProcessName(_processId, _machineName);
+
+                        if (_processName == null)
```

Comment:
When `ProcessManager.GetProcessName` calls `Interop.Kernel32.GetProcessName` and passes the result to `NtProcessInfoHelper.GetProcessShortName(processName)`

https://github.com/dotnet/runtime/blob/38e3d142f3409c6e71dbf67860d0379940301be0/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs#L110-L111

which returns an empty string for empty spans:

https://github.com/dotnet/runtime/blob/21f9b736b7a915c80f8c4df74fac4905bb9bf2fd/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L422-L427

so `_processName` is never going to be `null` here?

Reply:
Thanks @adamsitnik . I have committed your suggested change.